vslib/SwitchState.cpp

```diff
@@ -199,6 +218,79 @@ void SwitchState::asyncOnLinkMsg(
     m_switchConfig->m_eventQueue->enqueue(std::make_shared<Event>(EVENT_TYPE_NET_LINK_MSG, payload));
 }
 
+sai_status_t SwitchState::getNetStat(
+        _In_ sai_stat_id_t counterId,
+        _In_ std::string& ifName,
+        _Out_ uint64_t& counter)
+{
+    SWSS_LOG_ENTER();
+
+    FILE *fp;
+    char stat[100];
+    std::string counterName;
+    sai_status_t rc = SAI_STATUS_SUCCESS;
+
+    try
+    {
+        counterName = m_statIdMap.at(counterId);
+    }
+    catch (std::out_of_range const& e)
+    {
+        /* not an error only few counters supported */
+        return rc;
+    }
+
+    std::string cmd = "/bin/cat /sys/class/net/" + ifName + "/statistics/" + counterName;
+    fp = popen(cmd.c_str(), "r");
+
+    if ((fp != NULL) && (fgets(stat, sizeof(stat), fp) != NULL))
+    {
+        counter = atoi(stat);
+        SWSS_LOG_DEBUG("%s [ %s ]  = %s", ifName.c_str(), counterName.c_str(), stat);
+    }
+    else
+    {
+        counter = -1;
+        SWSS_LOG_ERROR("get counter cmd %s failed fp=%x", cmd, fp);
+        rc = SAI_STATUS_FAILURE;
+    }
+
+    if (fp != NULL)
+    {
+        pclose(fp);
+    }
```

Comment:
as proposed, works for me:

```c++
#include <fstream>
#include <iostream>
#include <string>

// $ g++ test.cpp -o test
// $ ./test
// 40515

int main()
{

    std::string s = "/sys/class/net/lo/statistics/rx_bytes";

    std::ifstream ifs (s.c_str(), std::ifstream::in);

    uint64_t stat;

    if (ifs.good())
    {
        ifs >> stat;

        std::cout << stat << std::endl;
    }

    return 0;
}
```

Reply:
ok. May be "std::ios::in" or read was an issue.